### PR TITLE
Add recursive chainbreeding

### PR DESCRIPTION
**Pull request by dot-Comfey**

### CODE REVIEW

File: test/sim/team-validator/breeding.js [hunk 1]
```diff
@@ -203,17 +203,22 @@ describe('Team Validator', function () {
...
+		assert.legalTeam(team, 'gen5ou');
+
+		team = [
+			{species: 'corphish', ability: 'hypercutter', moves: ['dragondance', 'metalclaw'], evs: {hp: 1}},
+		];
+		assert.legalTeam(team, 'gen4ou');
 	});
 
-	it.skip('should accept this chainbreed on Toxicroak', function () {
+	it.skip('should reject Volbeat with both Lunge and Dizzy Punch in Gen 7', function () {
```
Comment (KrisXV) [inline test/sim/team-validator/breeding.js:218]: should this skip be removed?

Comment (dot-Comfey) [reply test/sim/team-validator/breeding.js:218]: This Volbeat still passes as legal when it shouldn't, and I don't think a fix for this has been coded yet.

Comment (DaWoblefet) [reply test/sim/team-validator/breeding.js:218]: This is just the diff being weird and you didn't actually touch the Volbeat test, right?

**Comment by DaWoblefet** [comment]
I am initially wary about recursive loops that can't be escaped - do we have a failsafe in case this somehow went on "too many times"?

File: test/sim/team-validator/breeding.js (hunk 1 above)
Comment (dot-Comfey) [reply test/sim/team-validator/breeding.js:218]: Yeah, I moved up the Toxicroak test to separate it from the skipped test.

**Comment by dot-Comfey** [comment]
I'm not sure how many iterations the validator can handle, but I made sure findEggMoveFathers is called by fatherCanLearn as little as possible, and pokemonBlacklist should ensure that fatherCanLearn is never called with the same parameters twice (Dragonite and Snorlax will ignore this but the egg group restrictions I added to fatherCanLearn will prevent infinite loops as well).

**Comment by dot-Comfey** [comment]
Now that I think about it, I think it's going to be exceptionally rare for a third iteration of findEggMoveFathers (validateSet -> findEggMoveFathers -> fatherCanLearn -> findEggMoveFathers -> fatherCanLearn -> findEggMoveFathers) to happen, so I suppose a limit of two iterations can work as a failsafe.

**Comment by MathyFurret** [comment]
I'll try to fuzz this with random egg move sets tomorrow

**Comment by MathyFurret** [comment]
My methodology was to validate random non-Field mons with 1-4 random egg moves in gen 5. I only tested 900 sets, many more could be teseted if needed. [Here](https://pastebin.com/mpZbRXhc) were the discrepancies I found between master and this PR.

I don't have the patience to manually test all of these, but here are the few that I did:
Ledyba with Dizzy Punch + Encore: **New version is correct.** Breed Encore onto Spinda and teach it Dizzy Punch. Breed both moves onto Volbeat and then onto Ledyba
Larvitar with Curse + Pursuit + Stomp: **New version is correct.** Breed Stomp and Curse onto Rhyhorn (via Smeargle), then breed both onto Cranidos and teach it Pursuit, then breed all onto Larvitar
Bagon with Dragon Rush + Thrash + Dragon Rage + Defense Curl: **New version is incorrect.** Correct me if I'm missing something but, while it can get the first 3 from Gible, there's now no way it can get Defense Curl.

**Comment by dot-Comfey** [comment]
Yeah Salamence is the problem here; Defense Curl is detected as legal due to being a Gen 3 tutor, but can't be from Gen 3 due to having an egg move from Gen 4. Looks like this is a problem in the current implementation as well, since Defense Curl is able to escape egg move validation.

**Comment by DaWoblefet** [comment]
Thanks dot-Comfey!

**Pull request merged**